packages/documentation/gatsby-node.js

```diff
@@ -8,69 +6,226 @@
+  }
 
+  result.data.allFile.edges.forEach(({ node }) => {
     const { headings, html } = node.childMarkdownRemark;
 
+    // Don't create pages for empty files.
+    if (!html) return;
+
+    const {
+      gitRemote: {
+        webLink: repositoryUrl,
+        ref: branch,
+      },
+      name,
+      relativeDirectory,
+      relativePath,
+      sourceInstanceName: source,
+    } = node;
+
+    // Use the top level heading as the title.
     const title = headings && headings[0] && headings[0].value;
 
-    // Link to GitHub document.
-    const sourceUrl = `${repositoryUrl}/blob/${branch}/${node.relativePath}`;
+    // Build the full link to the document.
+    const sourceUrl = `${repositoryUrl}/blob/${branch}/${relativePath}`;
 
     createPage({
-      path: `${node.relativeDirectory}/${node.name}`,
+      path: `${relativeDirectory}/${name}`,
       component: path.resolve('./src/templates/page.js'),
       context: {
         html,
+        source,
         sourceUrl,
         title,
       },
-    })
+    });
+  });
+};
+
+/**
+ * Derive and include the source URL on automatically generated pages,
+ * specifically from src/pages.
+ *
+ * External pages, such as those pulled from va.gov-team or formation-react,
+ * should already have sources and also source URLs associated with them.
+ */
+const setSourceUrl = ({ page, actions, reporter }) => {
+  if (page.context.sourceUrl) return;
+
+  // For debugging, log if an externally sourced page is missing the URL.
+  if (page.context.source) {
+    reporter.warn(`No source URL found for external page: ${page}`);
+    return;
+  }
+
+  // Check if this page is sourced from src/pages.
+  const { createPage, deletePage } = actions;
+  const match = page.componentPath && page.componentPath.match(
+    `/packages/documentation/src/pages/.+`
+  );
+
+  // Avoid setting source URLs that we can't confidently determine.
+  if (!match) return;
+
+  // Build the source URL once we've determined that the page is from src/pages.
+  const [relativePath] = match;
+  const sourceUrl = `${GITHUB_FILE_BASE_URL}${relativePath}`;
+
+  deletePage(page);
+
+  createPage({
+    ...page,
+    context: {
+      ...page.context,
+      sourceUrl,
+    },
+  });
+};
+
+
+exports.onCreateNode = ({ node, getNode, actions }) => {
+  const { createNodeField } = actions;
+  const parent = getNode(node.parent);
+
+  // Enable filtering MDX nodes by source.
+  if (node.internal.type === 'Mdx' && parent.sourceInstanceName) {
+    createNodeField({
+      node,
+      name: 'sourceInstanceName',
+      value: parent.sourceInstanceName,
+    });
+  }
+};
+
+exports.createPages = async helpers => {
+  await createFormationReactPages(helpers);
+  await createVaGovTeamPages(helpers);
+};
+
+exports.onCreatePage = helpers => {
+  setSourceUrl(helpers);
+};
+
+exports.onCreateWebpackConfig = ({ actions }) => {
+  actions.setWebpackConfig({
+    resolve: {
+      modules: [path.resolve(__dirname, 'src'), 'node_modules'],
+    },
```

Comment:
I pulled the stuff from the former `mdx-pages` plugin into here. Maybe it's cleaner to split into plugins, but not sure of best practice here. I just didn't think these functions did enough with the data to be considered "plugins".

Could be worth splitting out into plugins if we had more sources to pull from, but ultimately, I thought keeping our custom usage of all the Gatsby Node APIs in one place would be easier for discoverability.